skan/src/ui/NewItem.scala

```diff
@@ -9,79 +9,63 @@ import tui.widgets.tabs.TabsWidget
 
 object NewItem:
   def render(frame: Frame, state: NewItemState): Unit =
-    val chunks = Layout(
-      direction = Direction.Vertical,
-      margin = Margin(3),
-      constraints = Array(
-        Constraint.Length(2),
-        Constraint.Length(3),
-        Constraint.Length(3),
-        Constraint.Length(3),
-        Constraint.Length(3)
-      )
-    ).split(frame.size)
-
-    Header.render(frame, chunks(0))
-
-    val titleWidget = ParagraphWidget(
-      text = Text.nostyle(state.title),
-      block = Some(
-        BlockWidget(borders = Borders.ALL, title = Some(Spans.nostyle("Title")))
-      ),
-      style = (state.focusedInput, state.inputMode) match
-        case (InputSection.Title, InputMode.Input) =>
-          Style.DEFAULT.fg(Color.Yellow)
-        case _ => Style.DEFAULT
-    )
 
-    frame.renderWidget(titleWidget, chunks(1))
-
-    val descriptionWidget = ParagraphWidget(
-      text = Text.nostyle(state.description),
-      block = Some(
-        BlockWidget(
-          borders = Borders.ALL,
-          title = Some(Spans.nostyle("Description"))
+    val titleWidget = Widget: (area, buf) =>
+      BlockWidget(
+        borders = Borders.ALL,
+        title = Some(Spans.nostyle("Title"))
+      )(
+        ParagraphWidget(
+          text = Text.nostyle(state.title),
+          style = (state.focusedInput, state.inputMode) match
+            case (InputSection.Title, InputMode.Input) =>
+              Style.DEFAULT.fg(Color.Yellow)
+            case _ => Style.DEFAULT
+        )
+      ).render(area, buf)
+      if state.focusedInput == InputSection.Title && state.inputMode == InputMode.Input
+      then
+        frame.setCursor(
+          x = area.x + Grapheme(state.title).width + 1,
+          y = area.y + 1
         )
-      ),
-      style = (state.focusedInput, state.inputMode) match
-        case (InputSection.Description, InputMode.Input) =>
-          Style.DEFAULT.fg(Color.Yellow)
-        case _ => Style.DEFAULT
-    )
 
-    frame.renderWidget(descriptionWidget, chunks(2))
+    val descriptionWidget = Widget: (area, buf) =>
+      BlockWidget(
+        borders = Borders.ALL,
+        title = Some(Spans.nostyle("Description"))
+      )(
+        ParagraphWidget(
+          text = Text.nostyle(state.description),
+          style = (state.focusedInput, state.inputMode) match
+            case (InputSection.Description, InputMode.Input) =>
+              Style.DEFAULT.fg(Color.Yellow)
+            case _ => Style.DEFAULT
+        )
+      ).render(area, buf)
+      if state.focusedInput == InputSection.Description && state.inputMode == InputMode.Input
+      then
+        frame.setCursor(
+          x = area.x + Grapheme(state.description).width + 1,
+          y = area.y + 1
+        )
```

Comment:
So this is what I was referring to on the other PR, it's a but of a bummer since I only need to do this when I'm on the description and in input mode, but still need to wrap it in a `Widget` even when it's not that. It's not the worst, but yea.